general/gpu.py

```diff
@@ -27,22 +36,41 @@ def __init__(self):
     try:
       pynvml.nvmlInit()
       self.pynvmlLoaded = True
+      logger.info('pynvml initialized.')
     except Exception as e:
-      self.pynvmlLoaded = False
       logger.error('Could not init pynvml.' + str(e))
 
-    if self.pynvmlLoaded and pynvml.nvmlDeviceGetCount() > 0:
-      self.cudaDevicesFound = pynvml.nvmlDeviceGetCount()
+    if not self.pynvmlLoaded:
+      try:
+        amdsmi.amdsmi_init()
+        self.amdsmiLoaded = True
+        self.needRefresh = True # bug in amdsmi
```

Comment:
Old amdsmi can't refresh it's telemetry without reinit. New version of ROCm (6.1) fix that issue, but that version has no official support for pytorch yet.